File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowExecuteUseCase.java
package es.ull.project.application.usecase.workflow;

import java.net.URI;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

import es.ull.project.domain.entity.Execution;
import es.ull.project.domain.valueobject.NodeName;

public interface WorkFlowExecuteUseCase {

    /**
     * Executes the workflow according to the given SourceFlow input file and return the
     * output of the last node.
     * 
     * @param workflow
     * @param pendingUriFiles map of pending URI files to be processed by the
     *                        workflow nodes
     * @return execute data
     */
    public Optional<Execution> executeWorkFlow(UUID workFlowId, Map<NodeName, URI> pendingUriFiles);
}

File: front-end/src/application/usecase/workflow/workflow-execute-use-case.ts
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { NodeName } from '@/domain/value-object/node-name';
import { Execution } from '@/domain/entity/execution';

export interface WorkFlowExecuteUseCase {
  /**
   * Executes the workflow
   * and returns the output of the last node.
   *
   * @param workflowId - The UUID of the workflow to execute
   * @returns The resulting File of the last node
   */
  executeFlow(workFlowId: UllUUID, pendingUriFiles: Map<NodeName, string>): Promise<Either<DataError, Execution>>;
}
